lib/token.cpp

```diff
@@ -2523,10 +2525,27 @@ bool Token::hasKnownSymbolicValue(const Token* tok) const
     });
 }
 
+const ValueFlow::Value* Token::getKnownValue() const
+{
+    if (!mImpl->mValues)
+        return nullptr;
+    auto it = std::find_if(mImpl->mValues->begin(), mImpl->mValues->end(), std::mem_fn(&ValueFlow::Value::isKnown));
+    return it == mImpl->mValues->end() ? nullptr : &*it;
+}
```

Comment:
I don't think we should have such a function. It could lead to bugs where a different valueflow type is returned.

Reply:
Agreed.

Fortunately the subsequent cleanups made it unused and it can be removed.